### ISSUE TITLE
Lack of activity - maintenance needed. What's up?

### ISSUE THREAD

**Issue by tzaeru** [alexmingoia/koa-router#503]
Hi,

This is one of the most used middleware for Koa. Looking at the issue list and pull requests, it seems that there's however no maintenance activity going on.

In issue #494, this problem has been covered in brief. In the discussion, the current owner said that maintenance would continue beginning in February. That was over two months ago.

What's the current plan with the maintenance of this project? If it's not going to be actively maintained by the current owner, could it be moved under e.g. the KoaJS Github organization?

**Comment by crobinson42** [comment]
@tzaeru if you're not already in the Koa community slack, here's where to sign-up https://communityinviter.com/apps/koa-js/koajs

I'm curious about this too - there's needed maintenance on this package and it has gone stale. I know a few people have talked about forking and moving under koajs org for the community to maintain.

**Comment by spotshare-nick** [comment]
This package was sold to ZijianHe, and while he says he plans to maintain the package, he hasn't yet done anything.  Plenty of other issues discussing the sale and you can also see it in the commit history from the original author.

**Comment by beary** [comment]
@ZijianHe 如果你没有时间或者能力来维护的话，为何不将此项目转入 https://github.com/koajs 下呢？目前 npm 上最后一次版本发布都是一年前的事情了，占着此仓库却不做任何事情，有点损人不利己。

**Comment by spotshare-nick** [comment]
See issue #494.  He's only had it for 2 months, and he he paid to purchase the package.  I do agree that Koa should fork this package and start fixing things, but for the moment this is a rather lost cause until he starts committing regularly.

**Comment by srounce** [comment]
As the project is MIT licensed a fork is completely possible (under a 'koajs-community' org or something) though it'd only be worthwhile if there was the community drive to get on top of the maintenance backlog.

**Comment by o8e** [comment]
This should probably just be forked now 🙍‍♂

**Comment by sartrey** [comment]
See also [eggjs/egg-router](https://github.com/eggjs/egg-router) .

**Comment by vladshcherbin** [comment]
What a fucking joke, package sold to be dead instead of giving to someone who will actually maintain it. Router, one of core packages in koa apps 🤦‍♂

**Comment by divmgl** [comment]
I agree, this is pretty frustrating. Running into `param` and `prefix` bugs that are already have open pull requests but this package is unmaintained. Really considering switching to `express` or `hapi`.

**Comment by dead-horse** [comment]
@ZijianHe If you don't have time to maintain this package, I'll fork it to https://github.com/koajs and release as `@koa/router`, so we can maintain this package by community. :(

**Comment by fengmk2** [comment]
@dead-horse let's fork it!

**Comment by fl0w** [comment]
@dead-horse, @fengmk2: I think we (the Koa community) are done asking. It's MIT, let's just stick a fork in @koajs. I use it daily and can help maintain it. This is also a security matter at this point from my point of view.

**Comment by tzaeru** [comment]
> 
> 
> @ZijianHe If you don't have time to maintain this package, I'll fork it to https://github.com/koajs and release as `@koa/router`, so we can maintain this package by community. :(

Fork it for KoaJS, I'm also interested in participating in the occasional maintenance of this package. It's well structured package that is both simple and powerful. It's my go-to both at work and in my hobby projects.

**Comment by vladshcherbin** [comment]
What about package name, `koa-router` is quite popular and it would be nice to continue using it. [Npm dispute](https://www.npmjs.com/policies/disputes) can be opened, it fits at least one reason:

*Adopt an "abandoned" package*

Maybe this will also help understand current owner that this package is needed and hopefully transfer it to koa org. If no actions will follow, it will at least free the package name.

I can do this and then transfer to koa org or any koa org member.

**Comment by dead-horse** [comment]
forked to https://github.com/koajs/koa-router and released to npm as `@koa/router`. Anyone who want to contribute to it can create pull requests now. 😸

**Comment by thomas-darling** [comment]
@dead-horse, as mentioned by @vladshcherbin, the Koa organization should maybe also consider raising a dispute over the name `koa-router` on NPM. The way this package changed ownership, and the complete lack of maintenance or communication of any kind thereafter, does raise some serious concerns for those relying on it - many of whom may not be aware that anything changed.

**Comment by jamesaspence** [comment]
Just came here to say this is awesome, and I'm really happy there is an official router. One thing I'd like to point out is that https://github.com/koajs/koa/wiki has no mention whatsoever of this official fork happening. We should get that updated!

**Comment by ralyodio** [comment]
What's the story? I'm starting a koa project, is this project dead now?

**Comment by jamesaspence** [comment]
@chovy please use the official version here: https://github.com/koajs/koa-router. It's been forked.
